error for build: Undefined symbols for architecture arm64:   _png_do_expand_palette_rgb8_neon

```

command like this:

% cmake -G "Xcode" ..
% cmake --build . 


and the error has been appeared:


```


 
```

file: error("Failed to open diagnostics file") (in target 'dnn_imagenet_train_ex' from project 'examples')

Ld /Users/xxx/Downloads/dlib-master/examples/build/Release/dnn_metric_learning_on_images_ex normal (in target 'dnn_metric_learning_on_images_ex' from project 'examples')
    cd /Users/xxx/Downloads/dlib-master/examples
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ -target arm64-apple-macos12.3 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk -L/Users/xxx/Downloads/dlib-master/examples/build/Release -F/Users/xxx/Downloads/dlib-master/examples/build/Release -filelist /Users/xxx/Downloads/dlib-master/examples/build/examples.build/Release/dnn_metric_learning_on_images_ex.build/Objects-normal/arm64/dnn_metric_learning_on_images_ex.LinkFileList -Wl,-search_paths_first -Wl,-headerpad_max_install_names /Users/xxx/Downloads/dlib-master/examples/build/dlib_build/Release/libdlib.a /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/lib/libcblas.tbd /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/lib/liblapack.tbd /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/lib/libsqlite3.tbd -Xlinker -no_adhoc_codesign -Xlinker -dependency_info -Xlinker /Users/xxx/Downloads/dlib-master/examples/build/examples.build/Release/dnn_metric_learning_on_images_ex.build/Objects-normal/arm64/dnn_metric_learning_on_images_ex_dependency_info.dat -o /Users/xxx/Downloads/dlib-master/examples/build/Release/dnn_metric_learning_on_images_ex
Undefined symbols for architecture arm64:
  "_png_do_expand_palette_rgb8_neon", referenced from:
      _png_do_read_transformations in libdlib.a(pngrtran.o)
  "_png_do_expand_palette_rgba8_neon", referenced from:
      _png_do_read_transformations in libdlib.a(pngrtran.o)
  "_png_riffle_palette_neon", referenced from:
      _png_do_read_transformations in libdlib.a(pngrtran.o)
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)


** BUILD FAILED **


The following build commands failed:
	Ld /dlib-master/examples/build/Release/dnn_metric_learning_on_images_ex normal (in target 'dnn_metric_learning_on_images_ex' from project 'examples')
```



* **Version**: <!-- What version of dlib? -->
> the lastest master code

* **Where did you get dlib**: <!-- Did you get it from an official source like dlib.net, this github repo, or somewhere else? -->
> from GitHub get glib source code

* **Platform**: <!-- include something like `lsb_release -a` output, or if Windows, version and 32-bit or
  64-bit -->
> Darwin Kernel Version 21.5.0: Tue Apr 26 20:57:23 PDT 2022; root:xnu-8020.121.3~2/RELEASE_ARM64_T8110 arm64


* **Compiler**: <!-- What compiler are you using?  What version of that compiler? -->
> Apple clang version 13.1.6 (clang-1316.0.21.2.5
> cmake version 3.24.1

--- a/dlib/CMakeLists.txt
+++ b/dlib/CMakeLists.txt
@@ -441,6 +441,7 @@ if (NOT TARGET dlib)
             set(source_files ${source_files}
                external/libpng/arm/arm_init.c
                external/libpng/arm/filter_neon_intrinsics.c
+               external/libpng/arm/palette_neon_intrinsics.c
                external/libpng/png.c
                external/libpng/pngerror.c
                external/libpng/pngget.c

